pykit/types.py

```diff
@@ -0,0 +1,157 @@
+class Parameter:
+    def __init__(self, parameterType, direction='inout', byValue=False, isReturn=False):
```

Comment:
The way dbsock handles return values is weird and unintuitive. A return value _by definition_ cannot be an input, so why does it have a direction? I think there should have been a separate way to specify the return value instead of trying to shoe-horn it in as a "parameter".

I'd like to see us hide this quirk as much as possible. Can we split out the return value handling in to a separate class, maybe `ReturnValue` and then add a new method `add_return`?